### PR TITLE
Make error message explicit in legend.py

### PR DESCRIPTION
**Pull request by ClementWalter**
## PR Summary

Update the error message when label is a string and starts with `_` so as to make it clearer to the user.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -404,9 +404,9 @@ def __init__(self, parent, handles, labels,
         _lab, _hand = [], []
         for label, handle in zip(labels, handles):
             if isinstance(label, str) and label.startswith('_'):
-                cbook._warn_external('The handle {!r} has a label of {!r} '
-                                     'which cannot be automatically added to'
-                                     ' the legend.'.format(handle, label))
+                cbook._warn_external('The label {!r} of handle {!r} cannot be '
+                                     'a string starting with '
+                                     '"_"'.format(handle, label))
```
Comment (QuLogic) [inline lib/matplotlib/legend.py:409]: These arguments need to be flipped now.
```suggestion
                                     '"_"'.format(label, handle))
```

**Review by QuLogic — DISMISSED** [review]
Please try to give your commits meaningful messages.

**Comment by tacaswell** [comment]
@ClementWalter Would you be willing to add a test for this?

**Comment by ClementWalter** [comment]
ok i will try to dit in the next few days

**Comment by QuLogic** [comment]
Ping?

**Comment by jklymak** [comment]
This needs a rebase.  Prob doesn't desperately need a test, thought it would be nice to up our code coverage.

**Comment by ClementWalter** [comment]
Hi!

Sorry I was doing completely other things I’ll fix this over the weekend promise

Thanks 

Clément Walter
+336.29.44.21.67

File: lib/matplotlib/legend.py [hunk 2]
```diff
@@ -404,9 +404,9 @@ def __init__(self, parent, handles, labels,
         _lab, _hand = [], []
         for label, handle in zip(labels, handles):
             if isinstance(label, str) and label.startswith('_'):
-                cbook._warn_external('The handle {!r} has a label of {!r} '
-                                     'which cannot be automatically added to'
-                                     ' the legend.'.format(handle, label))
+                cbook._warn_external('The label {!r} of handle {!r} cannot be '
+                                     'a string starting with '
+                                     '"_"'.format(label, handle))
```
Comment (timhoffm) [inline lib/matplotlib/legend.py:409]: The merge conflict comes from internal changes `cbook._warn_external` -> `_api.warn_external`.
While you're at it, using f-strings makes this more readable.

```suggestion
                _api.warn_external(f'The label {label!r} of handle {handle!r} '
                                   'cannot be a string starting with "_"')

**Pull request closed without merging**